### PR TITLE
prune filter handling

### LINKED ISSUE #14182: `podman system prune --filter="label!=key"` doesn't work
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When calling ``podman system prune --filter="label!=com.test.test"`` I'm getting error output:

```
Error: label! is an invalid filter
```

**Steps to reproduce the issue:**

1. Call `podman system prune --filter="label!=com.test.test"`

**Describe the results you received:**

```
WARNING! This command removes:
	- all stopped containers
	- all networks not used by at least one container
	- all dangling images
	- all dangling build cache

Are you sure you want to continue? [y/N] y
Error: label! is an invalid filter
```

**Describe the results you expected:**

I should be able to prune all the images and containers which doesn't contain given label.

**Additional information you deem important (e.g. issue happens only occasionally):**

Happens all the time.

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.0.3
API Version:  4.0.3
Go Version:   go1.18
Built:        Fri Apr  1 20:21:54 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.3
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 12
  distribution:
    distribution: fedora
    variant: silverblue
    version: "36"
  eventLogger: journald
  hostname: loki.packetseekers.eu
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.5-300.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 5937229824
  memTotal: 33382322176
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.4.4-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.4
      commit: 6521fcc5806f20f6187eb933f9f45130c86da230
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8584425472
  swapTotal: 8589930496
  uptime: 10h 33m 12.19s (Approximately 0.42 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /var/home/jkonecny/.config/containers/storage.conf
  containerStore:
    number: 17
    paused: 0
    running: 4
    stopped: 13
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.8.1-3.fc36.x86_64
      Version: |-
        fusermount3 version: 3.10.5
        fuse-overlayfs: version 1.8.1
        FUSE library version 3.10.5
        using FUSE kernel interface version 7.31
  graphRoot: /var/home/jkonecny/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 741
  runRoot: /run/user/1000/containers
  volumePath: /var/home/jkonecny/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.3
  Built: 1648837314
  BuiltTime: Fri Apr  1 20:21:54 2022
  GitCommit: ""
  GoVersion: go1.18
  OsArch: linux/amd64
  Version: 4.0.3
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.0.3-1.fc36.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Fedora 36 but it happened also on Fedora 35.

### PR DESCRIPTION
**Pull request by cdoern**
network and container prune could not handle the label!=... filter. vendor in c/common to fix this and
add some podman level handling to make everything run smoothly

resolves #14182

Signed-off-by: Charlie Doern <cdoern@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
--label=filter!=key now works as expected
```

### CODE REVIEW

**Comment by cdoern** [comment]
this vendors in c/common thats why it is so large. a few PRs do this ATM so one will get in first!

**Comment by rhatdan** [comment]
LGTM
/approve
@containers/podman-maintainers PTAL

File: test/e2e/prune_test.go
Comment (vrothberg) [inline test/e2e/prune_test.go:292]: I think we should check the output here as well.  At the moment, we only know that it succeeded.

**Review by vrothberg — COMMENTED** [review]
Just one nit. Nice work, @cdoern !

**Review by flouthoc — APPROVED** [review]
LGTM other than one comment above.

File: test/e2e/prune_test.go
Comment (cdoern) [reply test/e2e/prune_test.go:292]: done

**Comment by cdoern** [comment]
can we cancel the hold on this? or does anyone have objections @containers/podman-maintainers

**Comment by Luap99** [comment]
LGTM

/hold cancel

**Pull request merged**
